crates/ef-testing/src/models/case.rs

```diff
@@ -100,35 +207,11 @@ impl Case for BlockchainTestCase {
 
     async fn run(&self) -> Result<(), ef_tests::Error> {
         let env = Arc::new(KakarotTestEnvironmentContext::from_dump_state().await);
-        let test = self.tests.clone();
-
-        // Prepare the pre test state
-        let env_binding = env.clone();
-        let _ = tokio::task::spawn_blocking(move || {
-            let kakarot = env_binding.kakarot();
-            let kakarot_address = kakarot.kakarot_address;
-
-            // Get lock on the Starknet sequencer
-            let mut starknet = env_binding
-                .sequencer()
-                .sequencer
-                .backend
-                .state
-                .blocking_write();
-
-            let eoa_class_hash = get_eoa_class_hash(env_binding.clone(), &starknet)
-                .expect("failed to get eoa class hash");
-            let class_hashes = ClassHashes::new(
-                kakarot.proxy_class_hash,
-                eoa_class_hash,
-                kakarot.contract_account_class_hash,
-            );
-            write_test_state(&test, kakarot_address, class_hashes, &mut starknet)
-        })
-        .await
-        .map_err(|err| ef_tests::Error::Assertion(err.to_string()))?;
+        // handle pretest
+        let _handled_pre_state = self.handle_pre_state(&env).await?;
```

Comment:
some notes: 

using `let _` bindings, even when you end a call with the `?`, will result in the error messages not getting conveyed up

Reply:
I tried running this and it does panick in the main so this looks ok
```
fn foo() -> Result<(), std::io::Error> {
    Err(std::io::Error::new(std::io::ErrorKind::Other, "oh no!"))
}

fn bar() -> Result<(), std::io::Error> {
    let _ = foo()?;
    Ok(())
}

fn main() {
    bar().unwrap()
}
```